include/vctr/Expressions/ExpressionTemplate.h

```diff
@@ -155,28 +155,36 @@ void tryApplyingRuntimeArgsToSrc (const RuntimeArgs& args, Src& src)
     It expects that the expression class has a suitable value_type defined before this, that the second
     class template type name is SrcType and that there is a private member SrcType src.
 
-    It defines an Expression type, a constructor, a getStorageInfo(), a size() and an isNotAliased (const void*)
-    member function according to the expression template conventions.
+    It defines the member srcName of type SrcType, an Expression type, a constructor, a getStorageInfo(),
+    a size(), an isNotAliased (const void*) and a iterateOverRuntimeArgChain (const RuntimeArgs&) member
+    function according to the expression template conventions.
+
+    Note that this macro contains a private and public sections and ends with the public access specifier.
+    Therefore all code following this macro will be in the public section if not explicitly specified
+    differently.
  */
 // clang-format off
-#define VCTR_COMMON_UNARY_EXPRESSION_MEMBERS(ExpressionName)                               \
-using Expression = ExpressionTypes<value_type, SrcType>;                                   \
-                                                                                           \
-template <class Src>                                                                       \
-constexpr ExpressionName (Src&& s) : src (std::forward<Src> (s)) {}                        \
-                                                                                           \
-constexpr const auto& getStorageInfo() const { return src.getStorageInfo(); }              \
-                                                                                           \
-constexpr size_t size () const { return src.size (); }                                     \
-                                                                                           \
-constexpr bool isNotAliased (const void* other) const { return src.isNotAliased (other); } \
-                                                                                           \
-template <size_t i, class RuntimeArgs>                                                     \
-void iterateOverRuntimeArgChain (const RuntimeArgs& rtArgs)                                \
-{                                                                                          \
-    tryApplyingRuntimeArgsToThisExpression<i> (rtArgs, *this);                             \
-    tryApplyingRuntimeArgsToSrc<i + 1> (rtArgs, src);                                      \
-}
+#define VCTR_COMMON_UNARY_EXPRESSION_MEMBERS(ExpressionName, srcName)                          \
+private:                                                                                       \
+    using Expression = ExpressionTypes<value_type, SrcType>;                                   \
+    SrcType srcName;                                                                           \
```

Comment:
What is this member used fore?

Reply:
I'd propose you to have a look at the expression readme for a better overview: https://github.com/sonible/VCTR/blob/f0175800f9defff312c1bc104215462a78746b79/include/vctr/Expressions/Readme.md 

Short answer: This holds the source of this expression, it could either be a reference to the source container or a chained source expression

Reply:
Thank you, I'll check it out!
What I meant though is that it seems you introduced a new member that is not referred to anywhere else in the diff

Reply:
This is the replacement for all those explicit members that have now been removed in all expressions: 
![image](https://user-images.githubusercontent.com/13800135/230037071-7dab1aca-10a9-42d1-97d3-f6cd5af5d88e.png)

Reply:
But you haven't changed its name to `srcName` anywhere. Or is this merely a member that exists but is not needed anywhere atm?

Reply:
Also... wouldn't one expect a variable `srcName` to contain the name of a source?

Reply:
Thx for explaining, I was confused for a moment. Makes sense now.